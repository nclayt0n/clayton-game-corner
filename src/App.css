/* http://meyerweb.com/eric/tools/css/reset/ 
   v2.0 | 20110126
   License: none (public domain)
*/

html,
body,
div,
span,
applet,
object,
iframe,
h1,
h2,
h3,
h4,
h5,
h6,
p,
blockquote,
pre,
a,
abbr,
acronym,
address,
big,
cite,
code,
del,
dfn,
em,
img,
ins,
kbd,
q,
s,
samp,
small,
strike,
strong,
sub,
sup,
tt,
var,
b,
u,
i,
center,
dl,
dt,
dd,
ol,
ul,
li,
fieldset,
form,
label,
legend,
table,
caption,
tbody,
tfoot,
thead,
tr,
th,
td,
article,
aside,
canvas,
details,
embed,
figure,
figcaption,
footer,
header,
hgroup,
menu,
nav,
output,
ruby,
section,
summary,
time,
mark,
audio,
video {
    margin: 0;
    padding: 0;
    border: 0;
    font-size: 100%;
    font: inherit;
    vertical-align: baseline;
}


/* HTML5 display-role reset for older browsers */

article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
menu,
nav,
section {
    display: block;
}

body {
    line-height: 1;
}

ol,
ul {
    list-style: none;
}

blockquote,
q {
    quotes: none;
}

blockquote:before,
blockquote:after,
q:before,
q:after {
    content: '';
    content: none;
}

table {
    border-collapse: collapse;
    border-spacing: 0;
}

:root {
    --mainColor: rgb(17, 72, 82);
}

img {
    width: 200px;
    height: 200px;
}

main {
    margin-bottom: 150px;
}

nav {
    background-color: white;
}

article,
section {
    display: flex;
    flex-direction: column;
    text-align: center;
    border: 1px solid black;
    width: 300px;
    margin: 80px auto;
    padding: 15px;
}

p {
    padding: 10px;
    margin: auto;
}

textarea {
    border: 1px solid black;
}

footer {
    position: fixed;
    bottom: 0;
    display: flex;
    width: 100%;
    background-color: white;
}

h2,
h1 {
    font-size: 22px;
    text-align: center;
}

h3 {
    margin: 20px auto;
    text-align: center;
}

footer address {
    margin: auto;
}

form {
    display: flex;
    flex-direction: column;
    flex: 1;
}

form textarea {
    height: 150px;
}

button {
    border: 1px solid black;
    margin: auto;
    width: 150px;
    background-color: var(--mainColor);
    color: white;
}

header {
    position: fixed;
    top: 0;
    width: 100%;
    text-align: center;
    width: 100%;
    background-color: var(--mainColor);
}

aside {
    margin: 120px auto 20px;
}

#headerLogo {
    height: 90px;
    width: 80px;
    margin: 10px;
}

#loginContainer {
    margin-top: 150px;
}

#loginForm {
    width: 300px;
}

#loginForm input {
    width: 200px;
}

.notFoundPage {
    margin-top: 150px;
}