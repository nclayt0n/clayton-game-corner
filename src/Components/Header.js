import React from 'react';
import {withRouter} from 'react-router-dom';
import logo from '../images/smallCGCLogo.png';
class Header extends React.Component{
    render(){
        return(
        <>
            {(this.props.location.pathname==='/')
                ?''
                :(<header>
                    <img id='headerLogo'src={logo} alt='clayton game corner logo'/>
                </header>)
            }
        </>)
    }
}
export default withRouter(Header);