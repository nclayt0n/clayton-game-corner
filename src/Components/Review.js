import React from 'react';
import fillerImage from '../images/githubicon.png';

class Review extends React.Component{
    // createImages(review){
    //     return JSON.parse(review.picture).map((pic,idx)=>{
    //         return <img src={pic} key={idx} alt={`images from the game ${review.title}`}/>
    //     });
    // }
    render(){
        // let pics=this.createImages(this.props.review);
        return(
           <article>
                <h3>{this.props.review.title}</h3>
                <div>
                    {/* {pics} */}
                </div>
                <p>{this.props.review.review}
                </p>
                <p>
                    <a href={this.props.review.link} target='_blank' rel='noopener noreferrer'>Buy</a>
                </p>
            </article>)
    }
}
export default Review;