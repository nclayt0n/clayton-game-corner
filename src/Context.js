import React from 'react';

const Context = React.createContext({
    user_id: '',
    reviews: [],
    bio: '',
    upcomingGames: [],
    picture: '',
    addPicture: () => {},
    addReviews: () => {},
    addReview: () => {},
    deleteReview: () => {},
    updateReview: () => {},
    updateUpcomingGame: () => {},
    addBio: () => {},
    updateBio: () => {},
    addUpcomingGames: () => {},
    deleteUpcomingGame: () => {},
    addUpcomingGame: () => {},
});
export default Context;