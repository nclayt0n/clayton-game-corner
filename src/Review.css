.review-info {
    border: 1px solid var(--dark);
    width: 85%;
    margin: 20px auto 50px;
    padding: 20px;
    letter-spacing: .75px;
    border-radius: 4px;
    display: flex;
    flex-direction: column;
    background-color: var(--dark);
    color: white;
}

.review-list {
    width: 90%;
    margin: 130px auto 10px;
}

.review-info h4 {
    margin: 10px auto;
    font-size: 26px;
}

.review-info p {
    margin: 10px auto;
    width: 85%;
}

.review-info a {
    text-align: center;
    color: white;
    cursor: pointer;
    display: flex;
    justify-content: center;
}

.review-header {
    width: 75%;
    margin: auto;
    border-radius: 4px;
    padding: 10px;
}

.review-header h3 {
    color: var(--dark);
    font-size: 34px;
    font-weight: bold;
    text-align: center;
    margin: 5px;
    line-height: 40px;
}

.review {
    background-color: var(--medium);
    border: 1px solid var(--mainColor);
    padding: 20px;
}

.review-header div {
    margin: 15px auto 15px;
    border-top: 2px solid var(--medium);
}

.admin-game-info {
    padding: 30px;
    margin: 10px auto;
    width: 400px;
}

.admin-add-game-review {
    padding: 30px;
    margin: 150px auto 50px;
    width: 400px;
}

.admin-add-game-review legend {
    display: none;
}

.admin-game-info legend {
    font-size: 22px;
    font-weight: bold;
    margin: 10px auto;
}

.admin-game-info form {
    background-color: var(--dark);
    border: 2px solid var(--mainColor);
    padding: 20px;
    border-radius: 4px;
    color: white;
    margin: 10px;
    text-align: center;
    height: 365px;
}

.admin-add-game-review form {
    display: flex;
    flex-direction: column;
    background-color: var(--bright);
    border: 2px solid var(--bright);
    padding: 20px;
    border-radius: 4px;
    color: white;
    margin: 10px;
    text-align: center;
}

.admin-game-info fieldset,
.admin-add-game-review fieldset {
    margin: 20px auto;
}

.admin-game-info input,
.admin-game-info select,
.admin-game-info label,
.admin-game-info textarea,
.admin-add-game-review input,
.admin-add-game-review select,
.admin-add-game-review label,
.admin-add-game-review textarea {
    margin: 5px auto;
    width: 200px;
    line-height: 24px;
    border-radius: 4px;
    flex: 1;
    background-color: white;
    text-align: center;
}

.admin-game-info textarea,
.admin-add-game-review textarea {
    height: 150px;
    text-align: left;
    padding: 0;
}

.admin-game-info label,
.admin-add-game-review label {
    margin: 0 auto;
    text-align: left;
}

.admin-game-info input,
.admin-add-game-review input {
    color: var(--mainColor);
}

.admin-game-info select,
.admin-add-game-review select {
    border: 1px solid white;
    border-radius: 4px;
    height: 24px;
    color: var(--mainColor);
}

.admin-add-game-review button {
    background-color: var(--dark);
    border: 1px solid var(--dark);
    color: white;
    font-weight: bold;
}

.admin-game-info button {
    margin: 5px auto;
    background-color: var(--medium);
    border: 1px solid var(--medium);
    color: white;
    font-weight: bold;
}

#admin-review-header {
    width: 55%;
    margin: auto;
    border-radius: 4px;
    padding: 10px;
}

#admin-review-header h3 {
    color: var(--dark);
    font-size: 34px;
    font-weight: bold;
    text-align: center;
    margin: 5px;
}

#admin-review-header div {
    margin: 15px auto 15px;
    border-top: 2px solid var(--medium);
}

@media screen and (min-width:700px) {
    .review-list {
        margin: 200px auto 10px;
        width: 500px;
    }
    .review-info {
        width: 450px;
    }
    #admin-review-header {
        width: 35%;
    }
    #admin-review-header h3 {
        font-size: 34px;
    }
}