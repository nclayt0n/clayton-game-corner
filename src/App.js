import React from 'react';
import {Switch,Route} from 'react-router-dom';
import Context from './Context';
import LandingPage from './Components/LandingPage';
import PublicOnlyRoute from './Utils/PublicOnlyRoute';
import PrivateRoute from './Utils/PrivateRoute';
import TabletopGameReview from './Components/TabletopGameReviews';
import VideoGameReview from './Components/VideoGameReview';
import UpcomingGameList from './Components/UpcomingGameList';
import './App.css';
import NotFound from './Components/NotFound';
import Header from './Components/Header';
import Nav from './Components/Nav';
import Footer from './Components/Footer';
import UpcomingGameInfo from './Components/UpcomingGameInfo';
import AdminLandingPage from './AdminComponents/AdminLandingPage';
import AdminReview from './AdminComponents/AdminReviewList';
import AdminReviewList from './AdminComponents/AdminReviewList';
import AdminLogin from './AdminComponents/AdminLogin';
import AdminUpcomingGameList from './AdminComponents/AdminUpcomingGameList';
import TokenService from './services/token-service'

class App extends React.Component {
  constructor(){
    super()
      this.state={
          user_id:0,
          reviews:[],
          bio:'',
          upcomingGames:[]
      };
    }
    handleAddReviews=(reviews)=>{
    this.setState({
      reviews:[...reviews]
    })
  }
  handleAddUpcomings=(upcomingGames)=>{
    this.setState({
      upcomingGames:[...upcomingGames]
    })
  }
  handleAddBio=(bio)=>{
    this.setState({
      bio:bio
    })
  }
  render(){
    let userId;
    (TokenService.getAuthToken()===null)?userId=0
      :userId=TokenService.decodeAuthToken(TokenService.getAuthToken());
    
    const contextValue={
      user_id:userId,
      bio:this.state.bio,
      upcomingGames:this.state.upcomingGames,
      reviews:this.state.reviews,
      addReviews:this.handleAddReviews,
      addUpcomingGames:this.handleAddUpcomings,
      addBio:this.handleAddBio
    };
  return (
    <main className='App'>
        <Context.Provider value={contextValue}>
        <Switch>
          <Route exact path='/' component={LandingPage}/>
          <Route path='/game/review/tabletop' component={TabletopGameReview}/>
          <Route path='/game/review/video' component={VideoGameReview}/>
          <Route exact path='/game/upcoming' component={UpcomingGameList}/>
          <Route exact path='/game/upcoming/info' component={UpcomingGameInfo}/>
          <Route path='/nav' component={Nav}/>
          <PrivateRoute exact path='/admin' component={AdminLandingPage}/>
          <PrivateRoute exact path='/admin/game/review-list' component={AdminReviewList}/>
          <PrivateRoute exact path='/admin/game/upcoming-list' component={AdminUpcomingGameList}/>
          <Route exact path='/admin/login' component={AdminLogin}/>
          <Route component={NotFound}/>
        </Switch>
        </Context.Provider>
      <Footer/>
    </main>
  );
  }
}

export default App;