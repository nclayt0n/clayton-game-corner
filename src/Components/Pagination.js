import React from 'react';
import Context from '../Context';

class Pagination extends React.Component{
    static contextType=Context;
    
    render(){
        return(
            <section className='pagination-button-container'>
                {this.props.page<1 
                    ? null
                    :<button 
                        aria-label='button for previous page'
                        className='button' 
                        type='button' 
                        onClick={()=>this.props.setPage(this.props.page-1)}
                        >Back
                    </button>}

                <button 
                        aria-label='current page'
                        className='button' 
                        id='currentPage' 
                        type='button'>Page {this.props.page+1}
                    </button>
                {this.props.items.length <this.props.pageLimit
                    ?null
                    :
                    <button 
                        aria-label='button for next page of users'
                        className='button' 
                        type='button' 
                        onClick={()=>this.props.setPage(this.props.page+1)}>Next
                    </button>
                    }
            </section>
        )
    }
}
export default Pagination;
