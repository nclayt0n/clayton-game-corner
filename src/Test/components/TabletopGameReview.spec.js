import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter,StaticRouter} from 'react-router-dom';
import App from '../../App';
import Pagination from '../../Components/Pagination';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Review from '../../Components/Review';
import Context from '../../Context';
import TabletopGameReview from '../../Components/TabletopGameReviews';
import Header from '../../Components/Header';

describe('Tabletop ',()=>{
  it('renders without crashing', () => {
      let reviews = [{
            "title": "Test Review 1",
            "review": "This was the best game ever!",
            "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
            "link": "http://www.notArealSite.com",
            "game_type": "tabletop"
        },
        {
            "title": "Test Review 2",
            "review": "This was an ok game.",
            "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
            "link": "http://www.notArealSite.com",
            "game_type": "tabletop"
        }];
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <App>
                        <TabletopGameReview key={6}>
                            <Header/>
                            <h2>Tabletop Game Review</h2>
                                {reviews.map(review=>{
                                    return <Review key={28} review={review}/>
                                })}
                            <Pagination page={0} 
                                        pageLimit={10}
                                        items={reviews}/>
                        </TabletopGameReview>
                    </App>
                </BrowserRouter>,
                div
              );
            ReactDOM.unmountComponentAtNode(div);
        });
    it('snapshot', () => {
        let reviews = [{
            "title": "Test Review 1",
            "review": "This was the best game ever!",
            "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
            "link": "http://www.notArealSite.com",
            "game_type": "tabletop"
        },
        {
            "title": "Test Review 2",
            "review": "This was an ok game.",
            "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
            "link": "http://www.notArealSite.com",
            "game_type": "tabletop"
        }];
        const wrapper=shallow(
            <StaticRouter>
                <App>
                    <Context.Provider>
                        <TabletopGameReview>
                            <Header/>
                            <h2>Tabletop Game Review</h2>
                            {   reviews.map(review=>{
                                    return <Review key={7} review={review}/>
                                })}
                            <Pagination page={0} 
                                        pageLimit={10}
                                        items={reviews}/>
                        </TabletopGameReview>
                    </Context.Provider>
                </App>
            </StaticRouter>)
            expect(toJson(wrapper))
            .toMatchSnapshot()
    });
})