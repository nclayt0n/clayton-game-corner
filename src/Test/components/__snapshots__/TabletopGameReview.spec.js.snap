// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabletop  snapshot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  staticContext={Object {}}
>
  <withRouter(App)>
    <ContextProvider>
      <TabletopGameReview>
        <Header />
        <withRouter(Nav) />
        <h2>
          Tabletop Game Review
        </h2>
        <Review
          key="7"
          review={
            Object {
              "game_type": "tabletop",
              "link": "http://www.notArealSite.com",
              "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
              "review": "This was the best game ever!",
              "title": "Test Review 1",
            }
          }
        />
        <Review
          key="7"
          review={
            Object {
              "game_type": "tabletop",
              "link": "http://www.notArealSite.com",
              "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
              "review": "This was an ok game.",
              "title": "Test Review 2",
            }
          }
        />
        <Pagination
          items={
            Array [
              Object {
                "game_type": "tabletop",
                "link": "http://www.notArealSite.com",
                "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
                "review": "This was the best game ever!",
                "title": "Test Review 1",
              },
              Object {
                "game_type": "tabletop",
                "link": "http://www.notArealSite.com",
                "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
                "review": "This was an ok game.",
                "title": "Test Review 2",
              },
            ]
          }
          page={0}
          pageLimit={10}
        />
      </TabletopGameReview>
    </ContextProvider>
  </withRouter(App)>
</Router>
`;
